### PR TITLE
feat(): Implementing the Greenhouse connector

### PR DESCRIPTION
**Pull request by mohan-pogala**
# Purpose
This PR is to implement the Greenhouse connector for the fidesops
# Changes
The PR contains the YAML changes for both config and dataset
-
# Checklist

- [ ] Applicable documentation updated (guides, quickstart, postman collections, tutorial, fidesdemo, [database diagram](https://github.com/ethyca/fidesops/blob/main/docs/fidesops/docs/development/update_erd_diagram.md))
- [ ] Good unit test/integration test coverage
- [ ] This PR contains a DB migration. If checked, the reviewer should confirm with the author that the [down_revision correctly references the previous migration](https://ethyca.github.io/fidesops/development/contributing_details/#alembic-migrations) before merging
- [ ] The `Run Unsafe PR Checks` label has been applied, and checks have passed, if this PR touches any external services

# Ticket

Fixes #

### CODE REVIEW

File: data/saas/config/greenhouse_config.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+saas_config:
+  fides_key: greenhouse_connector_example
+  name: Greenhouse SaaS Config
+  description: A sample schema representing the Greenhouse connector for Fidesops
+  version: 0.0.1
+
+  connector_params:
+    - name: user_id
+    - name: username
+    - name: email
```
Comment (galvana) [inline data/saas/config/greenhouse_config.yml:10]: Remove email identity, same reasoning as the Gorgias connector https://github.com/ethyca/fidesops/pull/304/files#r830290394

File: data/saas/dataset/greenhouse_dataset.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                data_categories: [system.operations]
+          - name: last_name
+            data_categories: [user.provided.identifiable.name]
+            fidesops_meta:
+              data_type: string
+          - name: id
+            data_categories: [user.derived.identifiable.unique_id]
+            fidesops_meta:
+              primary_key: True
+              # Has to change the datatype to DOUBLE, currently only string is supporting
```
Comment (galvana) [inline data/saas/dataset/greenhouse_dataset.yml:36]: Can use `int` here.

**Review by galvana — COMMENTED** [review]
We can start testing this after we implement headers for the SaaS connectors, and the minor comments are addressed. Almost there!

File: data/saas/config/greenhouse_config.yml (hunk 1 above)
Comment (mohan-pogala) [reply data/saas/config/greenhouse_config.yml:10]: Done

File: data/saas/dataset/greenhouse_dataset.yml (hunk 2 above)
Comment (mohan-pogala) [reply data/saas/dataset/greenhouse_dataset.yml:36]: Done

**Pull request closed without merging**